From 9b793f1c7e69f9e7ddb71fbff36e7eb469188b8f Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Tue, 13 Dec 2022 10:44:19 +0000
Subject: [PATCH] Never allow shell login for ipmi created users

---
 user_mgr.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index f4a90ec..a7a9162 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -1515,7 +1515,7 @@ void UserMgr::executeUserAdd(const char* userName, const char* groups,
     // set EXPIRE_DATE to 0 to disable user, PAM takes 0 as expire on
     // 1970-01-01, that's an implementation-defined behavior
     executeCmd("/usr/sbin/useradd", userName, "-G", groups, "-m", "-N", "-s",
-               (sshRequested ? "/bin/sh" : "/sbin/nologin"), "-e",
+               (sshRequested ? "/bin/true" : "/bin/true"), "-e",
                (enabled ? "" : "1970-01-01"));
 }
 
@@ -1536,7 +1536,7 @@ void UserMgr::executeUserModify(const char* userName, const char* newGroups,
                                 bool sshRequested)
 {
     executeCmd("/usr/sbin/usermod", userName, "-G", newGroups, "-s",
-               (sshRequested ? "/bin/sh" : "/sbin/nologin"));
+               (sshRequested ? "/bin/true" : "/bin/true"));
 }
 
 void UserMgr::executeUserModifyUserEnable(const char* userName, bool enabled)
-- 
2.36.1

