From 396c7901816fe65f0363033e9fa0913529034ab3 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Mon, 13 Dec 2021 14:09:49 +0000
Subject: [PATCH] NCT7802: Add configuration via DTS

---
 drivers/hwmon/nct7802.c | 220 ++++++++++++++++++++++++++++++++++++----
 1 file changed, 203 insertions(+), 17 deletions(-)

diff --git a/drivers/hwmon/nct7802.c b/drivers/hwmon/nct7802.c
index d56f78327619..fd73723801ac 100644
--- a/drivers/hwmon/nct7802.c
+++ b/drivers/hwmon/nct7802.c
@@ -44,6 +44,8 @@ static const u8 REG_VOLTAGE_LIMIT_MSB_SHIFT[2][5] = {
 #define REG_PECI_ENABLE		0x23
 #define REG_FAN_ENABLE		0x24
 #define REG_VMON_ENABLE		0x25
+#define REG_FAN_TYPE		0x5e
+#define REG_FAN_MODE		0x5f
 #define REG_PWM(x)		(0x60 + (x))
 #define REG_SMARTFAN_EN(x)      (0x64 + (x) / 2)
 #define SMARTFAN_EN_SHIFT(x)    ((x) % 2 * 4)
@@ -79,8 +81,20 @@ struct nct7802_data {
 	struct mutex in_alarm_lock;
 };
 
+struct nct7802_platform_data {
+	s8 sensor_type[3];
+	s8 local_temp_enable;
+	s8 vcc_enable;
+	s8 vcore_enable;
+	s8 fan_enable[3];
+	s8 fan_dc[2];   /* direct current instead PWM */
+	s8 fan_od[3];   /* open drain */
+	s8 fan_inv[3];  /* inverted polarity */
+	s8 peci_enable[2];
+};
+
 static ssize_t temp_type_show(struct device *dev,
-			      struct device_attribute *attr, char *buf)
+				 struct device_attribute *attr, char *buf)
 {
 	struct nct7802_data *data = dev_get_drvdata(dev);
 	struct sensor_device_attribute *sattr = to_sensor_dev_attr(attr);
@@ -95,8 +109,8 @@ static ssize_t temp_type_show(struct device *dev,
 }
 
 static ssize_t temp_type_store(struct device *dev,
-			       struct device_attribute *attr, const char *buf,
-			       size_t count)
+				  struct device_attribute *attr, const char *buf,
+				  size_t count)
 {
 	struct nct7802_data *data = dev_get_drvdata(dev);
 	struct sensor_device_attribute *sattr = to_sensor_dev_attr(attr);
@@ -116,7 +130,7 @@ static ssize_t temp_type_store(struct device *dev,
 }
 
 static ssize_t pwm_mode_show(struct device *dev,
-			     struct device_attribute *attr, char *buf)
+				struct device_attribute *attr, char *buf)
 {
 	struct sensor_device_attribute *sattr = to_sensor_dev_attr(attr);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -168,7 +182,7 @@ static ssize_t pwm_store(struct device *dev, struct device_attribute *devattr,
 }
 
 static ssize_t pwm_enable_show(struct device *dev,
-			       struct device_attribute *attr, char *buf)
+				  struct device_attribute *attr, char *buf)
 {
 	struct nct7802_data *data = dev_get_drvdata(dev);
 	struct sensor_device_attribute *sattr = to_sensor_dev_attr(attr);
@@ -203,7 +217,7 @@ static ssize_t pwm_enable_store(struct device *dev,
 }
 
 static int nct7802_read_temp(struct nct7802_data *data,
-			     u8 reg_temp, u8 reg_temp_low, int *temp)
+				u8 reg_temp, u8 reg_temp_low, int *temp)
 {
 	unsigned int t1, t2 = 0;
 	int err;
@@ -356,7 +370,7 @@ static int nct7802_write_voltage(struct nct7802_data *data, int nr, int index,
 }
 
 static ssize_t in_show(struct device *dev, struct device_attribute *attr,
-		       char *buf)
+			    char *buf)
 {
 	struct sensor_device_attribute_2 *sattr = to_sensor_dev_attr_2(attr);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -388,7 +402,7 @@ static ssize_t in_store(struct device *dev, struct device_attribute *attr,
 }
 
 static ssize_t in_alarm_show(struct device *dev, struct device_attribute *attr,
-			     char *buf)
+				char *buf)
 {
 	struct sensor_device_attribute_2 *sattr = to_sensor_dev_attr_2(attr);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -509,8 +523,8 @@ static ssize_t fan_min_show(struct device *dev, struct device_attribute *attr,
 }
 
 static ssize_t fan_min_store(struct device *dev,
-			     struct device_attribute *attr, const char *buf,
-			     size_t count)
+				struct device_attribute *attr, const char *buf,
+				size_t count)
 {
 	struct sensor_device_attribute_2 *sattr = to_sensor_dev_attr_2(attr);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -694,7 +708,7 @@ static struct attribute *nct7802_temp_attrs[] = {
 };
 
 static umode_t nct7802_temp_is_visible(struct kobject *kobj,
-				       struct attribute *attr, int index)
+					  struct attribute *attr, int index)
 {
 	struct device *dev = kobj_to_dev(kobj);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -793,7 +807,7 @@ static struct attribute *nct7802_in_attrs[] = {
 };
 
 static umode_t nct7802_in_is_visible(struct kobject *kobj,
-				     struct attribute *attr, int index)
+					struct attribute *attr, int index)
 {
 	struct device *dev = kobj_to_dev(kobj);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -868,7 +882,7 @@ static struct attribute *nct7802_fan_attrs[] = {
 };
 
 static umode_t nct7802_fan_is_visible(struct kobject *kobj,
-				      struct attribute *attr, int index)
+					 struct attribute *attr, int index)
 {
 	struct device *dev = kobj_to_dev(kobj);
 	struct nct7802_data *data = dev_get_drvdata(dev);
@@ -1056,8 +1070,8 @@ static const struct regmap_config nct7802_regmap_config = {
 };
 
 static int nct7802_get_channel_config(struct device *dev,
-				      struct device_node *node, u8 *mode_mask,
-				      u8 *mode_val)
+					 struct device_node *node, u8 *mode_mask,
+					 u8 *mode_val)
 {
 	u32 reg;
 	const char *type_str, *md_str;
@@ -1102,7 +1116,7 @@ static int nct7802_get_channel_config(struct device *dev,
 
 	if (!strcmp(type_str, "voltage")) {
 		*mode_val |= (RTD_MODE_VOLTAGE & MODE_RTD_MASK)
-			     << MODE_BIT_OFFSET_RTD(reg - 1);
+				<< MODE_BIT_OFFSET_RTD(reg - 1);
 		*mode_mask |= MODE_RTD_MASK << MODE_BIT_OFFSET_RTD(reg - 1);
 		return 0;
 	}
@@ -1141,7 +1155,7 @@ static int nct7802_get_channel_config(struct device *dev,
 }
 
 static int nct7802_configure_channels(struct device *dev,
-				      struct nct7802_data *data)
+					 struct nct7802_data *data)
 {
 	/* Enable local temperature sensor by default */
 	u8 mode_mask = MODE_LTD_EN, mode_val = MODE_LTD_EN;
@@ -1160,6 +1174,151 @@ static int nct7802_configure_channels(struct device *dev,
 	return regmap_update_bits(data->regmap, REG_MODE, mode_mask, mode_val);
 }
 
+static int nct7802_parse_dt(struct device_node *node,
+				struct nct7802_platform_data *pdata)
+{
+	struct device_node *child;
+	const char *sen_type;
+	int ret;
+	int i;
+
+	for_each_child_of_node(node, child) {
+		u32 reg = 0xFF;
+
+		of_property_read_u32(child, "reg", &reg);
+		if (of_device_is_compatible(child, "nuvoton,nct7802-sensor")) {
+			ret = of_property_read_string(child,
+				              "sensor-type", &sen_type);
+			if (ret < 0) {
+				pr_err("%s: expected property sensor-type\n",
+						child->full_name);
+				return -EINVAL;
+			}
+			if (!strcmp(sen_type, "local")) {
+				pdata->local_temp_enable = 1;
+				continue;
+			} else if (!strcmp(sen_type, "vcc")) {
+				pdata->vcc_enable = 1;
+				continue;
+			} else if (!strcmp(sen_type, "vcore")) {
+				pdata->vcore_enable = 1;
+				continue;
+			}
+			if (reg >= ARRAY_SIZE(pdata->sensor_type)) {
+				pr_err("%s: invalid value: reg=%d\n",
+						child->full_name, reg);
+				return -EINVAL;
+			}
+			if (!strcmp(sen_type, "thermal-diode")) {
+				pdata->sensor_type[reg] = 1;
+			} else if (!strcmp(sen_type, "thermistor")) {
+				pdata->sensor_type[reg] = 2;
+			} else if (!strcmp(sen_type, "voltage")) {
+				pdata->sensor_type[reg] = 3;
+			} else {
+				pr_err("%s: invalid sensor-type=\"%s\"\n",
+						child->full_name, sen_type);
+				return -EINVAL;
+			}
+		} else if (of_device_is_compatible(child,
+							"nuvoton,nct7802-fan")) {
+			if (reg >= ARRAY_SIZE(pdata->fan_enable)) {
+				pr_err("%s: invalid value: %s=%d\n",
+						child->full_name, "reg", reg);
+				return -EINVAL;
+			}
+			pdata->fan_enable[reg] = 1;
+			if (of_property_read_bool(child, "direct-current")) {
+				if (reg >= ARRAY_SIZE(pdata->fan_dc)) {
+					pr_err("%s: invalid value: %s=%d\n",
+							child->full_name, "reg", reg);
+					return -EINVAL;
+				}
+				pdata->fan_dc[reg] = 1;
+			}
+			pdata->fan_od[reg] =
+				of_property_read_bool(child, "open-drain");
+			pdata->fan_inv[reg] =
+				of_property_read_bool(child, "invert");
+		} else if (of_device_is_compatible(child,
+						"nuvoton,nct7802-peci")) {
+			if (reg >= ARRAY_SIZE(pdata->peci_enable)) {
+				pr_err("%s: invalid value: reg=%d\n",
+						child->full_name, reg);
+				return -EINVAL;
+			}
+			pdata->peci_enable[reg] = 1;
+		}
+	}
+
+	return 0;
+}
+
+static int nct7802_platform_data_set(struct nct7802_data *data,
+				     struct nct7802_platform_data *pdata)
+{
+	int ret;
+	int i;
+
+	/* Enable ADC */
+	ret = regmap_update_bits(data->regmap, REG_START, 1, 1);
+	if (ret)
+		return ret;
+
+	for (i = 0; i < ARRAY_SIZE(pdata->sensor_type); i++) {
+		ret = regmap_update_bits(data->regmap, REG_MODE,
+				     3 << 2 * i,
+				     pdata->sensor_type[i] << 2 * i);
+		if (ret)
+			 return ret;
+	}
+	ret = regmap_update_bits(data->regmap, REG_MODE,
+				 1 << 6,
+				 pdata->local_temp_enable << 6);
+	if (ret)
+		return ret;
+	for (i = 0; i < ARRAY_SIZE(pdata->fan_enable); i++) {
+		ret = regmap_update_bits(data->regmap, REG_FAN_ENABLE,
+				     1 << i, pdata->fan_enable[i] << i);
+	}
+	for (i = 0; i < ARRAY_SIZE(pdata->peci_enable); i++) {
+		ret = regmap_update_bits(data->regmap, REG_PECI_ENABLE,
+				     1 << i, pdata->peci_enable[i] << i);
+		if (ret)
+			 return ret;
+	}
+	ret = regmap_update_bits(data->regmap, REG_VMON_ENABLE, 1,
+				 pdata->vcc_enable);
+	if (ret)
+		return ret;
+
+	ret = regmap_update_bits(data->regmap, REG_VMON_ENABLE, 1 << 1,
+				 pdata->vcore_enable << 1);
+	if (ret)
+		return ret;
+
+	for (i = 0; i < ARRAY_SIZE(pdata->fan_dc); i++) {
+		ret = regmap_update_bits(data->regmap, REG_FAN_TYPE, 1 << i,
+				     pdata->fan_dc[i] << i);
+		if (ret)
+			 return ret;
+	}
+	for (i = 0; i < ARRAY_SIZE(pdata->fan_od); i++) {
+		ret = regmap_update_bits(data->regmap, REG_FAN_MODE, 1 << i,
+				     pdata->fan_od[i] << i);
+		if (ret)
+			 return ret;
+	}
+	for (i = 0; i < ARRAY_SIZE(pdata->fan_inv); i++) {
+		ret = regmap_update_bits(data->regmap, REG_FAN_MODE,
+				     1 << (i + 4),
+				     pdata->fan_inv[i] << (i + 4));
+		if (ret)
+			return ret;
+	}
+	return 0;
+}
+
 static int nct7802_init_chip(struct device *dev, struct nct7802_data *data)
 {
 	int err;
@@ -1179,11 +1338,27 @@ static int nct7802_init_chip(struct device *dev, struct nct7802_data *data)
 
 static int nct7802_probe(struct i2c_client *client)
 {
+	struct nct7802_platform_data *pdata;
 	struct device *dev = &client->dev;
 	struct nct7802_data *data;
 	struct device *hwmon_dev;
 	int ret;
 
+	pdata = dev_get_platdata(dev);
+#ifdef CONFIG_OF
+	if (dev->of_node) {
+		pdata = devm_kzalloc(dev, sizeof(struct nct7802_platform_data),
+					GFP_KERNEL);
+		if (!pdata)
+			return -ENOMEM;
+		ret = nct7802_parse_dt(dev->of_node, pdata);
+		if (ret < 0) {
+			devm_kfree(dev, pdata);
+			return ret;
+		}
+	}
+#endif
+
 	data = devm_kzalloc(dev, sizeof(*data), GFP_KERNEL);
 	if (data == NULL)
 		return -ENOMEM;
@@ -1196,6 +1371,11 @@ static int nct7802_probe(struct i2c_client *client)
 	mutex_init(&data->in_alarm_lock);
 
 	ret = nct7802_init_chip(dev, data);
+	if (pdata)
+		ret = nct7802_platform_data_set(data, pdata);
+	else
+		ret = nct7802_init_chip(data);
+
 	if (ret < 0)
 		return ret;
 
@@ -1215,10 +1395,16 @@ static const struct i2c_device_id nct7802_idtable[] = {
 };
 MODULE_DEVICE_TABLE(i2c, nct7802_idtable);
 
+static const struct __maybe_unused of_device_id nct7802_dt_match[] = {
+	{ .compatible = "nuvoton,nct7802" },
+	{ },
+};
+
 static struct i2c_driver nct7802_driver = {
 	.class = I2C_CLASS_HWMON,
 	.driver = {
 		.name = DRVNAME,
+		.of_match_table = of_match_ptr(nct7802_dt_match),
 	},
 	.detect = nct7802_detect,
 	.probe_new = nct7802_probe,
-- 
2.33.1

