From d56ffd16d94cda991a6cbd2717d120fb81d21e70 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Wed, 7 Apr 2021 15:39:17 +0000
Subject: [PATCH] Replace space by underscore in sensor name

---
 src/sensorcommands.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sensorcommands.cpp b/src/sensorcommands.cpp
index fcc2593..f02d081 100644
--- a/src/sensorcommands.cpp
+++ b/src/sensorcommands.cpp
@@ -1465,7 +1465,7 @@ static int getSensorDataRecord(ipmi::Context::ptr ctx,
         name = path.substr(nameStart + 1, std::string::npos - nameStart);
     }
 
-    std::replace(name.begin(), name.end(), '_', ' ');
+    std::replace(name.begin(), name.end(), ' ', '_');
     if (name.size() > FULL_RECORD_ID_STR_MAX_LENGTH)
     {
         // try to not truncate by replacing common words
