From 97e8a735e000b0d98413d375a59bc90cc2d59cdf Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Thu, 27 Aug 2020 10:38:36 +0000
Subject: [PATCH] Instantiate-pca954x-mux-anyway

---
 drivers/i2c/muxes/i2c-mux-pca954x.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/muxes/i2c-mux-pca954x.c b/drivers/i2c/muxes/i2c-mux-pca954x.c
index 4ad6657..9aac1b9 100644
--- a/drivers/i2c/muxes/i2c-mux-pca954x.c
+++ b/drivers/i2c/muxes/i2c-mux-pca954x.c
@@ -481,7 +481,8 @@ static int pca954x_probe(struct i2c_client *client,
 	ret = pca954x_init(client, data);
 	if (ret < 0) {
 		dev_warn(dev, "probe failed\n");
-		return -ENODEV;
+		//return -ENODEV;
+               dev_warn(dev, "probe failed, instantiating anyway\n");
 	}
 
 	ret = pca954x_irq_setup(muxc);
-- 
2.7.4

