From 46934278294bfd35b4525d3d437de3367b993bf7 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Tue, 26 Jul 2022 14:46:17 +0000
Subject: [PATCH] lm75: Introduce postponed initialization

---
 drivers/hwmon/lm75.c | 26 ++++++++++++++++++++------
 1 file changed, 20 insertions(+), 6 deletions(-)

diff --git a/drivers/hwmon/lm75.c b/drivers/hwmon/lm75.c
index afdbb63237b9..932741e38c11 100644
--- a/drivers/hwmon/lm75.c
+++ b/drivers/hwmon/lm75.c
@@ -110,6 +110,7 @@ struct lm75_data {
 	unsigned int			sample_time;	/* In ms */
 	enum lm75_type			kind;
 	const struct lm75_params	*params;
+	bool	is_initialized;
 };
 
 /*-----------------------------------------------------------------------*/
@@ -318,6 +319,7 @@ static int lm75_write_config(struct lm75_data *data, u8 set_mask,
 	value = data->current_conf & ~clr_mask;
 	value |= set_mask;
 
+	data->is_initialized = false;
 	if (data->current_conf != value) {
 		s32 err;
 
@@ -327,6 +329,7 @@ static int lm75_write_config(struct lm75_data *data, u8 set_mask,
 			return err;
 		data->current_conf = value;
 	}
+	data->is_initialized = true;
 	return 0;
 }
 
@@ -335,7 +338,11 @@ static int lm75_read(struct device *dev, enum hwmon_sensor_types type,
 {
 	struct lm75_data *data = dev_get_drvdata(dev);
 	unsigned int regval;
-	int err, reg;
+	int err, reg, rc;
+
+	if (! data->is_initialized)
+		rc = lm75_write_config(data, data->params->set_mask,
+				data->params->clr_mask);
 
 	switch (type) {
 	case hwmon_chip:
@@ -377,7 +384,11 @@ static int lm75_write_temp(struct device *dev, u32 attr, long temp)
 {
 	struct lm75_data *data = dev_get_drvdata(dev);
 	u8 resolution;
-	int reg;
+	int reg, rc;
+
+	if (! data->is_initialized)
+		rc = lm75_write_config(data, data->params->set_mask,
+				data->params->clr_mask);
 
 	switch (attr) {
 	case hwmon_temp_max:
@@ -412,6 +423,11 @@ static int lm75_update_interval(struct device *dev, long val)
 	unsigned int reg;
 	u8 index;
 	s32 err;
+	int rc;
+
+	if (! data->is_initialized)
+		rc = lm75_write_config(data, data->params->set_mask,
+				data->params->clr_mask);
 
 	index = find_closest(val, data->params->sample_times,
 			     (int)data->params->num_sample_times);
@@ -617,16 +633,14 @@ static int lm75_probe(struct i2c_client *client)
 	/* Cache original configuration */
 	status = i2c_smbus_read_byte_data(client, LM75_REG_CONF);
 	if (status < 0) {
-		dev_dbg(dev, "Can't read config? %d\n", status);
-		return status;
+		dev_info(dev, "Unable to read status, instantiating anyway\n");
+		status = 0x60;
 	}
 	data->orig_conf = status;
 	data->current_conf = status;
 
 	err = lm75_write_config(data, data->params->set_mask,
 				data->params->clr_mask);
-	if (err)
-		return err;
 
 	err = devm_add_action_or_reset(dev, lm75_remove, data);
 	if (err)
-- 
2.36.0

