From 6daa567c757494811c23c2815f9f881003146e5c Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Tue, 22 Sep 2020 15:33:37 +0000
Subject: [PATCH 1/1] Revert Remove broadcast limitation of ipmb channel patch

---
 ipmbbridged.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ipmbbridged.cpp b/ipmbbridged.cpp
index 956e82a..d9e91a4 100644
--- a/ipmbbridged.cpp
+++ b/ipmbbridged.cpp
@@ -372,7 +372,8 @@ void IpmbChannel::processI2cEvent()
     }
 
     // if it is message received from ipmb channel, send out dbus signal
-    if (getChannelType() == ipmbChannelType::ipmb)
+    if (ipmbFrame->Header.Req.address == broadcastAddress &&
+        getChannelType() == ipmbChannelType::ipmb)
     {
         auto ipmbMessageReceived = IpmbRequest();
         ipmbMessageReceived.i2cToIpmbConstruct(ipmbFrame, r);
-- 
2.7.4

