diff --git a/redfish-core/lib/account_service.hpp b/redfish-core/lib/account_service.hpp
index e9a19a56..c946fb45 100644
--- a/redfish-core/lib/account_service.hpp
+++ b/redfish-core/lib/account_service.hpp
@@ -231,7 +231,7 @@ inline void parseLDAPConfigData(nlohmann::json& jsonResponse,
         BMCWEB_LOG_DEBUG << "Pushing the data groupName="
                          << obj.second.groupName << "\n";
 
-        nlohmann::json::array_t remoteGroupArray;
+        /*nlohmann::json::array_t remoteGroupArray;
         nlohmann::json::object_t remoteGroup;
         remoteGroup["RemoteGroup"] = obj.second.groupName;
         remoteGroupArray.emplace_back(std::move(remoteGroup));
@@ -241,7 +241,13 @@ inline void parseLDAPConfigData(nlohmann::json& jsonResponse,
         nlohmann::json::object_t localRole;
         localRole["LocalRole"] = getRoleIdFromPrivilege(obj.second.privilege);
         localRoleArray.emplace_back(std::move(localRole));
-        roleMapArray.emplace_back(std::move(localRoleArray));
+        roleMapArray.emplace_back(std::move(localRoleArray));*/
+
+        nlohmann::json::object_t remoteGroup;
+        remoteGroup["RemoteGroup"] = obj.second.groupName;
+        remoteGroup["LocalRole"] = getRoleIdFromPrivilege(obj.second.privilege);
+        roleMapArray.emplace_back(std::move(remoteGroup));
+
     }
 }
 
@@ -431,7 +437,8 @@ template <typename CallbackFunc>
 inline void getLDAPConfigData(const std::string& ldapType,
                               CallbackFunc&& callback)
 {
-
+    BMCWEB_LOG_DEBUG
+                << "OYYA:  getLDAPConfigData";
     const std::array<const char*, 2> interfaces = {ldapEnableInterface,
                                                    ldapConfigInterface};
 
@@ -1014,6 +1021,7 @@ inline void handleLDAPPatch(nlohmann::json& input,
                             const std::string& serverType)
 {
     std::string dbusObjectPath;
+    BMCWEB_LOG_ERROR << "OYYA: entry to check LDAP json";
     if (serverType == "ActiveDirectory")
     {
         dbusObjectPath = adConfigObject;
@@ -1044,6 +1052,7 @@ inline void handleLDAPPatch(nlohmann::json& input,
                              "ServiceEnabled", serviceEnabled,
                              "RemoteRoleMapping", remoteRoleMapData))
     {
+        BMCWEB_LOG_ERROR << "OYYA: cannot read JSON";
         return;
     }
 
@@ -1054,6 +1063,7 @@ inline void handleLDAPPatch(nlohmann::json& input,
     }
     if (ldapService)
     {
+        BMCWEB_LOG_ERROR << "OYYA: try to parse the input LDAP json";
         parseLDAPServiceJson(*ldapService, asyncResp, baseDNList,
                              userNameAttribute, groupsAttribute);
     }
@@ -1094,9 +1104,12 @@ inline void handleLDAPPatch(nlohmann::json& input,
                             const std::string& serverT) {
         if (!success)
         {
+            BMCWEB_LOG_DEBUG << "OYYA: getLDAPConfigData error ";
             messages::internalError(asyncResp->res);
             return;
         }
+        BMCWEB_LOG_DEBUG << "OYYA: we are here ";
+        //BMCWEB_LOG_DEBUG << "OYYA: userNameAttribute = " << userNameAttribute;
         parseLDAPConfigData(asyncResp->res.jsonValue, confData, serverT);
         if (confData.serviceEnabled)
         {
@@ -1412,6 +1425,7 @@ inline void handleAccountServicePatch(
 {
     if (!redfish::setUpRedfishRoute(app, req, asyncResp))
     {
+        BMCWEB_LOG_ERROR << "OYYA: Error in no route to redfish";
         return;
     }
     std::optional<uint32_t> unlockTimeout;
@@ -1429,6 +1443,7 @@ inline void handleAccountServicePatch(
             ldapObject, "ActiveDirectory", activeDirectoryObject, "Oem",
             oemObject))
     {
+        BMCWEB_LOG_ERROR << "OYYA: cannot read JSON";
         return;
     }
 
@@ -1438,6 +1453,7 @@ inline void handleAccountServicePatch(
             [asyncResp](const boost::system::error_code ec) {
             if (ec)
             {
+                BMCWEB_LOG_ERROR << "OYYA: error in minimal password length";
                 messages::internalError(asyncResp->res);
                 return;
             }
@@ -1456,6 +1472,7 @@ inline void handleAccountServicePatch(
 
     if (ldapObject)
     {
+        BMCWEB_LOG_ERROR << "OYYA: try to handle LDAP patch";
         handleLDAPPatch(*ldapObject, asyncResp, "LDAP");
     }
 
@@ -1477,6 +1494,7 @@ inline void handleAccountServicePatch(
 
     if (activeDirectoryObject)
     {
+
         handleLDAPPatch(*activeDirectoryObject, asyncResp, "ActiveDirectory");
     }
 
@@ -1486,6 +1504,7 @@ inline void handleAccountServicePatch(
             [asyncResp](const boost::system::error_code ec) {
             if (ec)
             {
+                BMCWEB_LOG_ERROR << "OYYA: unknown error:)";
                 messages::internalError(asyncResp->res);
                 return;
             }
@@ -1502,6 +1521,7 @@ inline void handleAccountServicePatch(
             [asyncResp](const boost::system::error_code ec) {
             if (ec)
             {
+                BMCWEB_LOG_ERROR << "OYYA: unknown error again";
                 messages::internalError(asyncResp->res);
                 return;
             }
diff --git a/redfish-core/lib/pcie.hpp b/redfish-core/lib/pcie.hpp
index 8af95806..5c5c172f 100644
--- a/redfish-core/lib/pcie.hpp
+++ b/redfish-core/lib/pcie.hpp
@@ -27,6 +27,8 @@
 #include <sdbusplus/unpack_properties.hpp>
 #include <utils/dbus_utils.hpp>
 
+/*This path is used to check OnBoard NIC for OCP devices*/
+
 namespace redfish
 {
 
@@ -196,11 +198,13 @@ inline void requestRoutesSystemPCIeDevice(App& app)
             const std::string* manufacturer = nullptr;
             const std::string* deviceType = nullptr;
             const std::string* generationInUse = nullptr;
+            const std::string* pnumber = nullptr;
+            const std::string* snumber = nullptr;
 
             const bool success = sdbusplus::unpackPropertiesNoThrow(
                 dbus_utils::UnpackErrorPrinter(), pcieDevProperties,
                 "Manufacturer", manufacturer, "DeviceType", deviceType,
-                "GenerationInUse", generationInUse);
+                "GenerationInUse", generationInUse, "PartNumber", pnumber, "SerialNumber", snumber);
 
             if (!success)
             {
@@ -229,6 +233,16 @@ inline void requestRoutesSystemPCIeDevice(App& app)
                 asyncResp->res.jsonValue["Manufacturer"] = *manufacturer;
             }
 
+            if (pnumber != nullptr)
+            {
+                asyncResp->res.jsonValue["PartNumber"] = *pnumber;
+            }
+
+            if (snumber != nullptr)
+            {
+                asyncResp->res.jsonValue["SerialNumber"] = *snumber;
+            }
+
             if (deviceType != nullptr)
             {
                 asyncResp->res.jsonValue["DeviceType"] = *deviceType;
@@ -240,6 +254,7 @@ inline void requestRoutesSystemPCIeDevice(App& app)
                 "/redfish/v1/Systems/system/PCIeDevices/" + device;
             asyncResp->res.jsonValue["Name"] = "PCIe Device";
             asyncResp->res.jsonValue["Id"] = device;
+            asyncResp->res.jsonValue["Type"] = "Mgmt LAN";
 
             asyncResp->res.jsonValue["PCIeFunctions"]["@odata.id"] =
                 "/redfish/v1/Systems/system/PCIeDevices/" + device +
diff --git a/redfish-core/lib/sensors.hpp b/redfish-core/lib/sensors.hpp
index c873e095..32b2754e 100644
--- a/redfish-core/lib/sensors.hpp
+++ b/redfish-core/lib/sensors.hpp
@@ -65,14 +65,14 @@ static auto sensorPaths = std::to_array<std::string_view>({
     "/xyz/openbmc_project/sensors/current",
     "/xyz/openbmc_project/sensors/airflow",
     "/xyz/openbmc_project/sensors/humidity",
-#ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
+//#ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
     "/xyz/openbmc_project/sensors/voltage",
     "/xyz/openbmc_project/sensors/fan_tach",
     "/xyz/openbmc_project/sensors/temperature",
     "/xyz/openbmc_project/sensors/fan_pwm",
     "/xyz/openbmc_project/sensors/altitude",
     "/xyz/openbmc_project/sensors/energy",
-#endif
+//#endif
     "/xyz/openbmc_project/sensors/utilization"
 });
 
