From 9acfc33ae46338e95cef30ce8bd9d3777c3b348a Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Mon, 8 Jun 2020 08:35:35 +0000
Subject: [PATCH 1/1] systemd: Reduce logging

---
 src/core/job.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/job.c b/src/core/job.c
index 9fe3035..484c66f 100644
--- a/src/core/job.c
+++ b/src/core/job.c
@@ -910,7 +910,7 @@ static void job_log_done_status_message(Unit *u, uint32_t job_id, JobType t, Job
         /* Show condition check message if the job did not actually do anything due to failed condition. */
         if ((t == JOB_START && result == JOB_DONE && !u->condition_result) ||
             (t == JOB_START && result == JOB_SKIPPED)) {
-                log_struct(LOG_INFO,
+                log_struct(LOG_DEBUG,
                            "MESSAGE=Condition check resulted in %s being skipped.", unit_status_string(u),
                            "JOB_ID=%" PRIu32, job_id,
                            "JOB_TYPE=%s", job_type_to_string(t),
-- 
2.7.4

