From 99963aa60c2236ac7affc5028ada1d80ace9d5c0 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Thu, 29 Sep 2022 08:34:40 +0000
Subject: [PATCH] Add ADM1278 device

---
 include/devices.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/devices.hpp b/include/devices.hpp
index c3f1985..0648304 100644
--- a/include/devices.hpp
+++ b/include/devices.hpp
@@ -115,6 +115,10 @@ const boost::container::flat_map<const char*, ExportTemplate, CmpStr>
           ExportTemplate("ads1015 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device",
                          createsHWMon::noHWMonDir)},
+         {"ADM1278",
+          ExportTemplate("adm1278 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
+                         "new_device", "delete_device",
+                         createsHWMon::noHWMonDir)},
          {"ADS7828",
           ExportTemplate("ads7828 $Address", "/sys/bus/i2c/devices/i2c-$Bus",
                          "new_device", "delete_device",
