From 349be89fa8cf81c0d9ea47abd6d27ba585c38152 Mon Sep 17 00:00:00 2001
From: Nikita Vedeneev <vedenena@yandex-team.ru>
Date: Thu, 1 Apr 2021 12:35:39 +0000
Subject: [PATCH] Move-to-PSUSensor

---
 src/PSUSensorMain.cpp | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 9e25608..4b97411 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -40,7 +40,7 @@
 
 static constexpr bool debug = false;
 
-static constexpr std::array<const char*, 22> sensorTypes = {
+static constexpr std::array<const char*, 34> sensorTypes = {
     "xyz.openbmc_project.Configuration.ADM1272",
     "xyz.openbmc_project.Configuration.ADM1278",
     "xyz.openbmc_project.Configuration.DPS800",
@@ -62,13 +62,27 @@ static constexpr std::array<const char*, 22> sensorTypes = {
     "xyz.openbmc_project.Configuration.RAA228000",
     "xyz.openbmc_project.Configuration.RAA228228",
     "xyz.openbmc_project.Configuration.RAA229004",
-    "xyz.openbmc_project.Configuration.TPS546D24"};
+    "xyz.openbmc_project.Configuration.TPS546D24",
+    "xyz.openbmc_project.Configuration.EMC1413",
+    "xyz.openbmc_project.Configuration.MAX31725",
+    "xyz.openbmc_project.Configuration.MAX31730",
+    "xyz.openbmc_project.Configuration.MAX6581",
+    "xyz.openbmc_project.Configuration.MAX6654",
+    "xyz.openbmc_project.Configuration.TMP112",
+    "xyz.openbmc_project.Configuration.TMP175",
+    "xyz.openbmc_project.Configuration.TMP421",
+    "xyz.openbmc_project.Configuration.TMP441",
+    "xyz.openbmc_project.Configuration.TMP75",
+    "xyz.openbmc_project.Configuration.NCT7802",
+    "xyz.openbmc_project.Configuration.JC42"};
 
 static std::vector<std::string> pmbusNames = {
     "adm1272",   "adm1278",   "dps800",    "ina219",   "ina230",   "isl68137",
     "isl68220",  "isl68223",  "isl69243",  "isl69260", "max16601", "max20710",
     "max20730",  "max20734",  "max20796",  "max34451", "pmbus",    "pxe1610",
-    "raa228000", "raa228228", "raa229004", "tps546d24"};
+    "raa228000", "raa228228", "raa229004", "tps546d24",
+    "emc1413", "max31725", "max31730","max6581", "max6654", "tmp112", "tmp175",
+    "tmp421", "tmp441", "tmp75", "nct7802", "jc42" };
 
 namespace fs = std::filesystem;
 
