From b1b14f5f9f142e1b22598b774bf9a832eb2c5b23 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Tue, 9 Jun 2020 13:01:03 +0000
Subject: [PATCH 1/1] Fix 0 instead of no reading

---
 src/NVMeSensor.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/NVMeSensor.cpp b/src/NVMeSensor.cpp
index 94b2354..875e816 100644
--- a/src/NVMeSensor.cpp
+++ b/src/NVMeSensor.cpp
@@ -244,7 +244,7 @@ void readAndProcessNVMeSensor(const std::shared_ptr<NVMeContext>& nvmeDevice)
             }
             else
             {
-                sensor->updateValue(0);
+                sensor->updateValue(std::numeric_limits<double>::quiet_NaN());
             }
 
             // cycle it back
-- 
2.7.4

