From 45fa0592a25c9a932453e618b0fec69e20ee4ae3 Mon Sep 17 00:00:00 2001
From: Nikita Vedeneev <vedenena@playground.rnd.yandex.net>
Date: Tue, 25 Aug 2020 14:54:50 +0000
Subject: [PATCH] Move-to-PSUSensor

---
 src/PSUSensorMain.cpp | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 937ca7c..9802c27 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -40,7 +40,7 @@
 
 static constexpr bool DEBUG = false;
 
-static constexpr std::array<const char*, 14> sensorTypes = {
+static constexpr std::array<const char*, 26> sensorTypes = {
     "xyz.openbmc_project.Configuration.ADM1272",
     "xyz.openbmc_project.Configuration.ADM1278",
     "xyz.openbmc_project.Configuration.INA219",
@@ -54,12 +54,26 @@ static constexpr std::array<const char*, 14> sensorTypes = {
     "xyz.openbmc_project.Configuration.MAX34451",
     "xyz.openbmc_project.Configuration.pmbus",
     "xyz.openbmc_project.Configuration.PXE1610",
-    "xyz.openbmc_project.Configuration.RAA228228"};
+    "xyz.openbmc_project.Configuration.RAA228228",
+    "xyz.openbmc_project.Configuration.EMC1413",
+    "xyz.openbmc_project.Configuration.MAX31725",
+    "xyz.openbmc_project.Configuration.MAX31730",
+    "xyz.openbmc_project.Configuration.MAX6581",
+    "xyz.openbmc_project.Configuration.MAX6654",
+    "xyz.openbmc_project.Configuration.TMP112",
+    "xyz.openbmc_project.Configuration.TMP175",
+    "xyz.openbmc_project.Configuration.TMP421",
+    "xyz.openbmc_project.Configuration.TMP441",
+    "xyz.openbmc_project.Configuration.TMP75",
+    "xyz.openbmc_project.Configuration.NCT7802",
+    "xyz.openbmc_project.Configuration.JC42"};
 
 static std::vector<std::string> pmbusNames = {
     "adm1272",  "adm1278",  "ina219",   "ina230",   "isl68137",
     "isl68220", "max16601", "max20730", "max20734", "max20796",
-    "max34451", "pmbus",    "pxe1610",  "raa228228"};
+    "max34451", "pmbus",    "pxe1610",  "raa228228",
+    "emc1413", "max31725", "max31730","max6581", "max6654", "tmp112", "tmp175",
+    "tmp421", "tmp441", "tmp75", "nct7802", "jc42" };
 
 namespace fs = std::filesystem;
 
