From ad721410a8d4242dd56f1c5cdbdbbeb4597f1089 Mon Sep 17 00:00:00 2001
From: Konstantin Klubnichkin <kitsok@yandex-team.ru>
Date: Mon, 19 Oct 2020 07:02:49 +0000
Subject: [PATCH] psusensor: Re-setup polling cycle in case of FD error

---
 src/PSUSensor.cpp | 92 +++++++++++++++++++++++++++++++++++++++----------------
 1 file changed, 66 insertions(+), 26 deletions(-)

diff --git a/src/PSUSensor.cpp b/src/PSUSensor.cpp
index a7a2619..f97edcd 100644
--- a/src/PSUSensor.cpp
+++ b/src/PSUSensor.cpp
@@ -155,8 +155,38 @@ void PSUSensor::setupRead(void)
         std::make_shared<boost::asio::streambuf>();
     std::weak_ptr<PSUSensor> weakRef = weak_from_this();
 
+    /* Check inputDev and open it if needed */
+    if (! inputDev.is_open())
+    {
+        std::cerr << __FUNCTION__ << " inputDev is not open, opening\n";
+        fd = open(path.c_str(), O_RDONLY);
+        if (fd > 0)
+        {
+            inputDev.assign(fd);
+        }
+        else
+        {
+            updateValue(std::numeric_limits<double>::quiet_NaN());
+            waitTimer.expires_from_now(boost::posix_time::milliseconds(sensorPollMs));
+            std::weak_ptr<PSUSensor> weakRef = weak_from_this();
+            waitTimer.async_wait([weakRef](const boost::system::error_code& ec) {
+                std::shared_ptr<PSUSensor> self = weakRef.lock();
+                if (ec == boost::asio::error::operation_aborted)
+                {
+                    // TODO FIXME Silent exit
+                    return; // we're being canceled
+                }
+                if (self)
+                {
+                    self->setupRead();
+                }
+            });
+            return;
+        }
+    }
+
     if (bridgeGpio.has_value())
-    {   
+    {
         (*bridgeGpio).request();
         // In case a channel has a bridge circuit,we have to turn the bridge on
         // prior to reading a value at least for one scan cycle to get a valid
@@ -169,11 +199,12 @@ void PSUSensor::setupRead(void)
                 std::shared_ptr<PSUSensor> self = weakRef.lock();
                 if (ec == boost::asio::error::operation_aborted)
                 {
+                    // TODO FIXME Silent exit
                     return; // we're being canceled
                 }
 
                 if (self)
-                {   
+                {
                     boost::asio::async_read_until(
                         self->inputDev, *buffer, '\n',
                         [weakRef, buffer](const boost::system::error_code& ec,
@@ -230,6 +261,7 @@ void PSUSensor::handleResponse(const boost::system::error_code& err)
             std::shared_ptr<PSUSensor> self = weakRef.lock();
             if (ec == boost::asio::error::operation_aborted)
             {
+                // TODO FIXME Silent exit
                 return;
             }
             if (self)
@@ -243,50 +275,58 @@ void PSUSensor::handleResponse(const boost::system::error_code& err)
     if ((err == boost::system::errc::bad_file_descriptor) ||
         (err == boost::asio::error::misc_errors::not_found))
     {
-        return;
+        /* Dilemma here:
+         * We can detect error and re-open device, but what to do if
+         * open fails? */
+        if (DEBUG) std::cerr << __FUNCTION__ << "Reading error occured " << err << "\n";
+        inputDev.close();
     }
-    std::istream responseStream(readBuf.get());
-    if (!err)
+    else
     {
-        std::string response;
-        try
+        std::istream responseStream(readBuf.get());
+        if (!err)
         {
-            std::getline(responseStream, response);
-            rawValue = std::stod(response);
-            responseStream.clear();
-            double nvalue = rawValue / sensorFactor;
+            std::string response;
+            try
+            {
+                std::getline(responseStream, response);
+                rawValue = std::stod(response);
+                responseStream.clear();
+                double nvalue = rawValue / sensorFactor;
 
-            updateValue(nvalue);
+                updateValue(nvalue);
 
-            if (minMaxReadCounter++ % 8 == 0)
+                if (minMaxReadCounter++ % 8 == 0)
+                {
+                    updateMinMaxValues();
+                }
+            }
+            catch (const std::invalid_argument&)
             {
-                updateMinMaxValues();
+                std::cerr << "Could not parse " << response << "\n";
+                incrementError();
             }
         }
-        catch (const std::invalid_argument&)
+        else
         {
-            std::cerr << "Could not parse " << response << "\n";
             incrementError();
         }
-    }
-    else
-    {
-        incrementError();
-    }
 
-    if (bridgeGpio.has_value())
-    {
-        (*bridgeGpio).release();
+        if (bridgeGpio.has_value())
+        {
+            (*bridgeGpio).release();
+        }
+
+        lseek(fd, 0, SEEK_SET);
     }
 
-    lseek(fd, 0, SEEK_SET);
     waitTimer.expires_from_now(boost::posix_time::milliseconds(sensorPollMs));
-
     std::weak_ptr<PSUSensor> weakRef = weak_from_this();
     waitTimer.async_wait([weakRef](const boost::system::error_code& ec) {
         std::shared_ptr<PSUSensor> self = weakRef.lock();
         if (ec == boost::asio::error::operation_aborted)
         {
+            // TODO FIXME Silent exit
             return;
         }
         if (self)
