From c1aea01a800bc3523e446dda4998c268c417a4bf Mon Sep 17 00:00:00 2001
From: Nikita Vedeneev <vedenena@yandex-team.ru>
Date: Mon, 5 Apr 2021 11:34:25 +0000
Subject: [PATCH] Add-nct7601-sensor

---
 src/PSUSensorMain.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 52ba91b..c1fdc0b 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -41,7 +41,7 @@
 #include <CLI/CLI.hpp>
 bool DEBUG = false;
 
-static constexpr std::array<const char*, 35> sensorTypes = {
+static constexpr std::array<const char*, 36> sensorTypes = {
     "xyz.openbmc_project.Configuration.ADM1272",
     "xyz.openbmc_project.Configuration.ADM1278",
     "xyz.openbmc_project.Configuration.DPS800",
@@ -76,7 +76,8 @@ static constexpr std::array<const char*, 35> sensorTypes = {
     "xyz.openbmc_project.Configuration.TMP75",
     "xyz.openbmc_project.Configuration.NCT7802",
     "xyz.openbmc_project.Configuration.JC42",
-    "xyz.openbmc_project.Configuration.LM5066I"};
+    "xyz.openbmc_project.Configuration.LM5066I",
+    "xyz.openbmc_project.Configuration.NCT7601"};
 
 static std::vector<std::string> pmbusNames = {
     "adm1272",   "adm1278",   "dps800",    "ina219",   "ina230",   "isl68137",
@@ -84,7 +85,7 @@ static std::vector<std::string> pmbusNames = {
     "max20730",  "max20734",  "max20796",  "max34451", "pmbus",    "pxe1610",
     "raa228000", "raa228228", "raa229004", "tps546d24",
     "emc1413", "max31725", "max31730","max6581", "max6654", "tmp112", "tmp175",
-    "tmp421", "tmp441", "tmp75", "nct7802", "jc42", "lm5066i" };
+    "tmp421", "tmp441", "tmp75", "nct7802", "jc42", "lm5066i", "nct7601" };
 
 namespace fs = std::filesystem;
 
