From 346acf8cc090693762bcd6d7bef2d1e373949c8e Mon Sep 17 00:00:00 2001
From: Nikita Vedeneev <vedenena@yandex-team.ru>
Date: Mon, 5 Apr 2021 11:54:00 +0000
Subject: [PATCH] Add-LM5066i05

---
 src/PSUSensorMain.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index c1fdc0b..e8897b1 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -41,7 +41,7 @@
 #include <CLI/CLI.hpp>
 bool DEBUG = false;
 
-static constexpr std::array<const char*, 36> sensorTypes = {
+static constexpr std::array<const char*, 37> sensorTypes = {
     "xyz.openbmc_project.Configuration.ADM1272",
     "xyz.openbmc_project.Configuration.ADM1278",
     "xyz.openbmc_project.Configuration.DPS800",
@@ -77,6 +77,7 @@ static constexpr std::array<const char*, 36> sensorTypes = {
     "xyz.openbmc_project.Configuration.NCT7802",
     "xyz.openbmc_project.Configuration.JC42",
     "xyz.openbmc_project.Configuration.LM5066I",
+    "xyz.openbmc_project.Configuration.LM5066I05",
     "xyz.openbmc_project.Configuration.NCT7601"};
 
 static std::vector<std::string> pmbusNames = {
@@ -85,7 +86,7 @@ static std::vector<std::string> pmbusNames = {
     "max20730",  "max20734",  "max20796",  "max34451", "pmbus",    "pxe1610",
     "raa228000", "raa228228", "raa229004", "tps546d24",
     "emc1413", "max31725", "max31730","max6581", "max6654", "tmp112", "tmp175",
-    "tmp421", "tmp441", "tmp75", "nct7802", "jc42", "lm5066i", "nct7601" };
+    "tmp421", "tmp441", "tmp75", "nct7802", "jc42", "lm5066i", "lm5066i05", "nct7601" };
 
 namespace fs = std::filesystem;
 
