diff --git a/src/allowlist-filter.cpp b/src/allowlist-filter.cpp
index aed090c..d1f7f7c 100644
--- a/src/allowlist-filter.cpp
+++ b/src/allowlist-filter.cpp
@@ -408,6 +408,25 @@ ipmi::Cc AllowlistFilter::filterMessage(ipmi::message::Request::ptr request)
     {
         if (!Allowlisted)
         {
+
+            if (request->ctx->netFn == 12 && request->ctx->cmd == 1) {
+                //The transport protocol cmd was received
+                log<level::INFO>("Openyard: The transport protocol cmd was received");
+                return ipmi::ccSuccess;
+            }
+
+            if (request->ctx->netFn == 10 && request->ctx->cmd == 17) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The FRU read cmd was received");
+                return ipmi::ccSuccess;
+            }
+
+            if (request->ctx->netFn == 10 && request->ctx->cmd == 16) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The FRU area request was received");
+                return ipmi::ccSuccess;
+            }
+            
             log<level::INFO>("Channel/NetFn/Cmd not Allowlisted",
                              entry("CHANNEL=0x%X", request->ctx->channel),
                              entry("NETFN=0x%X", int(request->ctx->netFn)),
@@ -455,6 +474,25 @@ ipmi::Cc AllowlistFilter::filterMessage(ipmi::message::Request::ptr request)
 
     if (!Allowlisted)
     {
+
+        if (request->ctx->netFn == 12 && request->ctx->cmd == 1) {
+                //The transport protocol cmd was received
+                log<level::INFO>("Openyard: The transport protocol cmd was received");
+                return ipmi::ccSuccess;
+        }
+
+        if (request->ctx->netFn == 10 && request->ctx->cmd == 17) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The FRU read cmd was received");
+                return ipmi::ccSuccess;
+        }
+
+        if (request->ctx->netFn == 10 && request->ctx->cmd == 16) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The FRU area request was received");
+                return ipmi::ccSuccess;
+        }
+        
         log<level::INFO>("Channel/NetFn/Cmd not allowlisted",
                          entry("CHANNEL=0x%X", request->ctx->channel),
                          entry("NETFN=0x%X", int(request->ctx->netFn)),
