diff --git a/include/oemcommands.hpp b/include/oemcommands.hpp
index 4f8166f..309981b 100644
--- a/include/oemcommands.hpp
+++ b/include/oemcommands.hpp
@@ -23,9 +23,13 @@ namespace ipmi
 namespace intel
 {
 
+/*OpenYard*/
+static constexpr NetFn netFnOpenyard = netFnOem;
+/*End*/
 static constexpr NetFn netFnGeneral = netFnOemOne;
 static constexpr NetFn netFnPlatform = netFnOemTwo;
 static constexpr NetFn netFnApp = netFnOemEight;
+//static constexpr NetFn netFnApp = netFnOem;
 
 namespace general
 {
@@ -97,13 +101,18 @@ static constexpr Cmd cmdClearCMOS = 0x91;
 
 namespace app
 {
-static constexpr Cmd cmdMdrStatus = 0x20;
-static constexpr Cmd cmdMdrComplete = 0x21;
+//static constexpr Cmd cmdMdrStatus = 0x20;
+/*OpenYard*/
+static constexpr Cmd cmdSmbiosReceived = 0x20;
+static constexpr Cmd cmdGetVgaHaredPtr = 0x21;
+/*End*/
+//static constexpr Cmd cmdMdrComplete = 0x21;
 static constexpr Cmd cmdMdrEvent = 0x22;
 static constexpr Cmd cmdMdrRead = 0x23;
 static constexpr Cmd cmdMdrWrite = 0x24;
 static constexpr Cmd cmdMdrLock = 0x25;
-static constexpr Cmd cmdMdrIIAgentStatus = 0x30;
+//static constexpr Cmd cmdMdrIIAgentStatus = 0x30;
+static constexpr Cmd cmdMdrIIAgentStatus = 0x21;
 static constexpr Cmd cmdMdrIIGetDir = 0x31;
 static constexpr Cmd cmdMdrIIGetDataInfo = 0x32;
 static constexpr Cmd cmdMdrIILockData = 0x33;
@@ -229,12 +238,14 @@ static constexpr const uint8_t targetInstanceShift = 1;
 enum class IPMINetfnIntelOEMAppCmd
 {
     mdrStatus = 0x20,
-    mdrComplete = 0x21,
+    ipmiOEMSendVgaSharedMemPtr = 0x21,
+    //mdrComplete = 0x21,
+    mdrComplete = 0x85,
     mdrEvent = 0x22,
     mdrRead = 0x23,
     mdrWrite = 0x24,
     mdrLock = 0x25,
-    mdr2AgentStatus = 0x30,
+    mdr2AgentStatus = 0x21,
     mdr2GetDir = 0x31,
     mdr2GetDataInfo = 0x32,
     mdr2LockData = 0x33,
diff --git a/include/smbiosmdrv2handler.hpp b/include/smbiosmdrv2handler.hpp
index 9bfb54f..bccdc9b 100644
--- a/include/smbiosmdrv2handler.hpp
+++ b/include/smbiosmdrv2handler.hpp
@@ -37,6 +37,7 @@ static constexpr const uint32_t mdr2SMSize =
     0x00100000; // Size of VGA share memory
 static constexpr const uint32_t mdr2SMBaseAddress =
     0x9FF00000; // Base address of VGA share memory
+    //0xAFF00000; // Base address of VGA share memory
 
 static constexpr const uint8_t mdrTypeII = 2; // MDR V2 type
 
diff --git a/ipmi-allowlist.conf b/ipmi-allowlist.conf
index 792e38f..6de4a6d 100644
--- a/ipmi-allowlist.conf
+++ b/ipmi-allowlist.conf
@@ -160,6 +160,8 @@
 0x2c:0x30:0x7f7f   //<Group Extension>:<Set TPM Configuration>
 0x2c:0x31:0x7f7f   //<Group Extension>:<TPM Configuration Update Complete>
 0x2c:0x37:0xff7f   //<Group Extension>:<Read PCIe Cable EEPROM Data>
+0x2e:0x20:0x7f7f   //<OpenYard>:<SMBIOS table received from host>
+0x2e:0x21:0x7f7f   //<OpenYard>:<Send Vgasharedmemory pointer to host>
 0x2e:0xC0:0x7f7f   //<Intel OEM>:<Enable/Disable Node Manager Policy Control>
 0x2e:0xC1:0x7f7f   //<Intel OEM>:<Set Node Manager Policy>
 0x2e:0xC2:0xff7f   //<Intel OEM>:<Get Node Manager Policy>
diff --git a/src/allowlist-filter.cpp b/src/allowlist-filter.cpp
index d1f7f7c..02e4793 100644
--- a/src/allowlist-filter.cpp
+++ b/src/allowlist-filter.cpp
@@ -402,6 +402,11 @@ ipmi::Cc AllowlistFilter::filterMessage(ipmi::message::Request::ptr request)
                        : first < value;
         });
 
+    if (request->ctx->netFn == 46 && request->ctx->cmd == 33) {
+        //The Storage protocol
+        log<level::INFO>("Openyard: The SMBIOS pointer was requested");
+    }
+    
     // no special handling for non-system-interface channels
     if (!(request->ctx->channel == ipmi::channelSystemIface ||
           request->ctx->channel == channelSMM))
@@ -426,6 +431,12 @@ ipmi::Cc AllowlistFilter::filterMessage(ipmi::message::Request::ptr request)
                 log<level::INFO>("Openyard: The FRU area request was received");
                 return ipmi::ccSuccess;
             }
+
+            if (request->ctx->netFn == 46 && request->ctx->cmd == 33) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The SMBIOS pointer was requested");
+                return ipmi::ccSuccess;
+            }
             
             log<level::INFO>("Channel/NetFn/Cmd not Allowlisted",
                              entry("CHANNEL=0x%X", request->ctx->channel),
@@ -492,6 +503,12 @@ ipmi::Cc AllowlistFilter::filterMessage(ipmi::message::Request::ptr request)
                 log<level::INFO>("Openyard: The FRU area request was received");
                 return ipmi::ccSuccess;
         }
+
+        if (request->ctx->netFn == 46 && request->ctx->cmd == 33) {
+                //The Storage protocol
+                log<level::INFO>("Openyard: The SMBIOS pointer was requested");
+                return ipmi::ccSuccess;
+        }
         
         log<level::INFO>("Channel/NetFn/Cmd not allowlisted",
                          entry("CHANNEL=0x%X", request->ctx->channel),
diff --git a/src/oemcommands.cpp b/src/oemcommands.cpp
index 11dab6f..ce14dd3 100644
--- a/src/oemcommands.cpp
+++ b/src/oemcommands.cpp
@@ -2895,6 +2895,23 @@ ipmi::RspType<> ipmiOEMSetFaultIndication(uint8_t sourceId, uint8_t faultType,
     return ipmi::responseSuccess();
 }
 
+/*OpenYard OEM cmd. Send the VGA shared memory pointer to BIOS*/
+/*ipmi::RspType<uint8_t,uint8_t,uint8_t,uint8_t,uint8_t,uint8_t,uint8_t> 
+      ipmiOEMSendVgaSharedMemPtr()
+{
+    uint8_t byte1, byte2, byte3, ptrByte1, ptrByte2, ptrByte3, ptrByte4;
+    phosphor::logging::log<phosphor::logging::level::INFO>("Openyard: The shared memory pointer request was received");
+    byte1 = 10;
+    byte2 = 60;
+    byte3 = 0;
+    ptrByte1 = 249;
+    ptrByte2 = 15;
+    ptrByte3 = 0;
+    ptrByte4 = 0;
+    return ipmi::responseSuccess(byte1, byte2, byte3, ptrByte1, ptrByte2, ptrByte3, ptrByte4);
+}*/
+/*End*/
+
 ipmi::RspType<uint8_t> ipmiOEMReadBoardProductId()
 {
     uint8_t prodId = 0;
@@ -4148,6 +4165,11 @@ static void registerOEMFunctions(void)
 
     registerHandler(prioOemBase, intel::netFnApp, intel::app::cmdPFRMailboxRead,
                     Privilege::Admin, ipmiOEMReadPFRMailbox);
+
+    /*OpenYard SMBIOS receive protocol handler register*/
+  //  registerHandler(prioOemBase, intel::netFnOpenyard, intel::general::cmdGetVgaHaredPtr,
+  //                  Privilege::Operator, ipmiOEMSendVgaSharedMemPtr);
+    /*end*/
 }
 
 } // namespace ipmi
diff --git a/src/smbiosmdrv2handler.cpp b/src/smbiosmdrv2handler.cpp
index c5d00a1..bcde8a4 100644
--- a/src/smbiosmdrv2handler.cpp
+++ b/src/smbiosmdrv2handler.cpp
@@ -32,6 +32,7 @@
 
 #include <cstdint>
 #include <fstream>
+#include <iostream>
 #include <string>
 #include <vector>
 
@@ -234,7 +235,7 @@ uint32_t MDRV2::calcChecksum32(uint8_t* buf, uint32_t len)
  *  - dirEntries
  *  - dataRequest
  */
-ipmi::RspType<uint8_t, uint8_t, uint8_t, uint8_t, uint8_t>
+/*ipmi::RspType<uint8_t, uint8_t, uint8_t, uint8_t, uint8_t>
     mdr2AgentStatus(uint16_t agentId, uint8_t dirVersion)
 {
     if (mdrv2 == nullptr)
@@ -270,7 +271,7 @@ ipmi::RspType<uint8_t, uint8_t, uint8_t, uint8_t, uint8_t>
 
     return ipmi::responseSuccess(mdrVersion, agentVersion, dirVersionResp,
                                  dirEntries, dataRequest);
-}
+}*/
 
 /** @brief implements mdr2 get directory command
  *  @param agentId
@@ -1271,16 +1272,50 @@ ipmi::RspType<> cmd_mdr2_data_done(uint16_t agentId, uint16_t lockHandle)
     return ipmi::responseSuccess();
 }
 
+/*OpenYard OEM cmd. Send the VGA shared memory pointer to BIOS*/
+ipmi::RspType<std::vector<uint8_t>> 
+      mdr2AgentStatus(uint8_t agentId, uint8_t data0, uint8_t data1)
+{
+    std::vector<uint8_t> raw_data( (uint8_t*)&mdr2SMBaseAddress
+                                   , (uint8_t*)&(mdr2SMBaseAddress) + sizeof(uint32_t));
+    return ipmi::responseSuccess(raw_data);
+}
+/*End*/
+
+//uint8_t dataSMbios[mdr2SMSize];
+
+/*OpenYard OEM cmd. Send the VGA shared memory pointer to BIOS*/
+ipmi::RspType<uint8_t,uint8_t> 
+      mdr2SmbiosReceived(uint16_t agentId, uint16_t data0, uint16_t data1, uint8_t data2)
+{
+    phosphor::logging::log<phosphor::logging::level::INFO>("Openyard: The SMBIOS table had received");
+    uint8_t byte2 = 60;
+    uint8_t byte3 = 0;
+
+    return ipmi::responseSuccess(byte2, byte3);
+}
+/*End*/
+
 static void register_netfn_smbiosmdrv2_functions(void)
 {
+    phosphor::logging::log<phosphor::logging::level::ERR>(
+            "OpenYard: MDR v2 functions registered");
     // MDR V2 Command
     // <Get MDRII Status Command>
-    ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnApp,
+    /*ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnApp,
+                          ipmi::intel::app::cmdMdrIIAgentStatus,
+                          ipmi::Privilege::Operator, mdr2AgentStatus);*/
+    
+    ipmi::registerOemHandler(ipmi::prioOemBase, ipmi::intel::netFnOpenyard,
                           ipmi::intel::app::cmdMdrIIAgentStatus,
                           ipmi::Privilege::Operator, mdr2AgentStatus);
 
+    ipmi::registerOemHandler(ipmi::prioOemBase, ipmi::intel::netFnOpenyard,
+                          ipmi::intel::app::cmdSmbiosReceived,
+                          ipmi::Privilege::Operator, mdr2SmbiosReceived);
+
     // <Get MDRII Directory Command>
-    ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnApp,
+    /*ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnApp,
                           ipmi::intel::app::cmdMdrIIGetDir,
                           ipmi::Privilege::Operator, mdr2GetDir);
 
@@ -1332,5 +1367,10 @@ static void register_netfn_smbiosmdrv2_functions(void)
     // <Send MDRII Data Done>
     ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnApp,
                           ipmi::intel::app::cmdMdrIIDataDone,
-                          ipmi::Privilege::Operator, cmd_mdr2_data_done);
+                          ipmi::Privilege::Operator, cmd_mdr2_data_done);*/
+    
+    // <Request a pointer to VGA shared memory>
+    /*ipmi::registerHandler(ipmi::prioOemBase, ipmi::intel::netFnOpenyard,
+                          ipmi::intel::app::cmdMdrComplete,
+                          ipmi::Privilege::Operator, ipmiOEMSendVgaSharedMemPtr);*/
 }
